pallets/block-rewards-hybrid/src/tests.rs

```diff
@@ -0,0 +1,530 @@
+    /// Increase in balances, based on `rewards` values, is verified.
+    ///
+    fn assert_distribution(&self, post_reward_state: &Self, rewards: &Rewards) {
+        assert_eq!(
+            self.treasury + rewards.treasury_reward,
+            post_reward_state.treasury
+        );
+        assert_eq!(
+            self.stakers + rewards.base_staker_reward + rewards.adjustable_staker_reward,
+            post_reward_state.stakers
+        );
+        assert_eq!(
+            self.collators + rewards.collators_reward,
+            post_reward_state.collators
+        );
+        assert_eq!(self.dapps + rewards.dapps_reward, post_reward_state.dapps);
+    }
+}
+
+/// Represents reward distribution balances for a single distribution.
+#[derive(PartialEq, Eq, Clone, RuntimeDebug)]
+struct Rewards {
+    treasury_reward: Balance,
+    base_staker_reward: Balance,
+    dapps_reward: Balance,
+    collators_reward: Balance,
+    adjustable_staker_reward: Balance,
+}
+
+impl Rewards {
+    /// Pre-calculates the reward distribution, using the provided `RewardDistributionConfig`.
+    /// Method assumes that total issuance will be increased by `BLOCK_REWARD`.
+    ///
+    /// Both current `total_issuance` and `TVL` are used. If these are changed after calling this function,
+    /// they won't be reflected in the struct.
+    ///
+    fn calculate(reward_config: &RewardDistributionConfig) -> Self {
+        // Calculate `tvl-independent` portions
+        let treasury_reward = reward_config.treasury_percent * BLOCK_REWARD;
+        let base_staker_reward = reward_config.base_staker_percent * BLOCK_REWARD;
+        let dapps_reward = reward_config.dapps_percent * BLOCK_REWARD;
+        let collators_reward = reward_config.collators_percent * BLOCK_REWARD;
+        let adjustable_reward = reward_config.adjustable_percent * BLOCK_REWARD;
+
+        // Calculate `tvl-dependent` portions
+        let total_issuance = <TestRuntime as Config>::Currency::total_issuance();
+        let tvl = <TestRuntime as Config>::DappsStakingTvlProvider::get();
+        let tvl_percentage = Perbill::from_rational(tvl, total_issuance);
+
+        // Calculate factor for adjusting staker reward portion
+        let factor = if reward_config.ideal_dapps_staking_tvl.is_zero() {
+            Perbill::one()
+        } else {
+            tvl_percentage / reward_config.ideal_dapps_staking_tvl
+        };
+
+        // Adjustable reward portions
+        let adjustable_staker_reward = factor * adjustable_reward;
+
+        Self {
+            treasury_reward,
+            base_staker_reward,
+            dapps_reward,
+            collators_reward,
+            adjustable_staker_reward,
+        }
+    }
+
+    fn sum(&self) -> Balance {
+        self.base_staker_reward
+            .saturating_add(self.adjustable_staker_reward)
+            .saturating_add(self.collators_reward)
+            .saturating_add(self.dapps_reward)
+            .saturating_add(self.treasury_reward)
```

Comment:
A minor thing, but for tests I'd suggest to go with good old `+` sign.

In case it overflows in tests, great, we've found a bug! 😄